library/client_events.js

```diff
@@ -114,6 +114,21 @@ module.exports = client => {
             client.createGuild(newMember.guild.id)
             client.log('Created new guild.')
           } else {
+            // if in any practice channel, member has rights to speak in practice room chat (can be muted)
+            let prChan = client.channels.find(chan => chan.name === 'practice-room-chat')
```

Comment:
I don't know discord.js well enough to know whether it makes sense to lift this code out and only run it once.

It seems also that `channels` reflects all the channels across all guilds. So if the production bot is invited to multiple guilds, there may be multiple channels with this name and this query will only return the first one, which may not be the right guild.

```javascript
chan => chan.name === 'practice-room-chat' && chan.guild.id === newMember.guild.id
```

Reply:
Oops. Actually I meant to get the channels property of the guild, not the channels property of the client, which saves needing to filter for this in the lambda. Thanks for catching this.